framework/src/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -110,10 +114,38 @@
      */
     protected final boolean sequentialSemantics;
 
+    /**
+     * Indicates that the signature inference is on.
+     */
+    private final boolean inferSignatures;
+
     public CFAbstractTransfer(CFAbstractAnalysis<V, S, T> analysis) {
         this.analysis = analysis;
         this.sequentialSemantics = !analysis.checker.hasOption("concurrentSemantics");
+        this.inferSignatures = analysis.checker.hasOption("inferSignatures");
+        if (inferSignatures) {
+            checkInvalidOptionsInferSignature(
+                    new String[]{"useDefaultsForUncheckedCode"});
```

Comment:
Making this an argument obscures the operation of the method. I would put that array somewhere cheap and refer directly to it in the method. Nobody calls this method with different arguments.